src/Solnet.Programs/Utilities/Serialization.cs

```diff
@@ -216,13 +216,23 @@ public static void WriteSingle(this byte[] data, float value, int offset)
         /// </summary>
         /// <param name="data">The data to encode.</param>
         /// <returns>The encoded data.</returns>
+        public static byte[] EncodeRustStringNoOffset(string data)
+        {
+            byte[] stringBytes = Encoding.UTF8.GetBytes(data);
+            byte[] encoded = new byte[stringBytes.Length];
+
+            encoded.WriteSpan(stringBytes, 0);
+
+            return encoded;
+        }
         public static byte[] EncodeRustString(string data)
         {
-            byte[] stringBytes = Encoding.ASCII.GetBytes(data);
-            byte[] encoded = new byte[stringBytes.Length + sizeof(uint)];
+            byte[] stringBytes = Encoding.UTF8.GetBytes(data);
+            byte[] encoded = new byte[stringBytes.Length+sizeof(ulong)];
+
+            encoded.WriteU64((ulong)stringBytes.Length, 0);

```

Comment:
If I am not mistaken dynamic length strings have 4 bytes which are used to define the string length, this should still remain as a u32 or not?

Reply:
I think this might be an array vs vec issue

Reply:
Or rather, a vec vs string issue?

Reply:
https://doc.rust-lang.org/stable/std/string/struct.String.html#representation

It has to have length and capacity, so 2 x u32, so you're right that u32 is the length, and that explains why I could only get the tx to go through with a u64

Reply:
I can't get it to work without it being a u64

I am not sure but it seems like this is something like what is happening where it's looking for a u64

https://github.com/solana-labs/solana/blob/e540b1cf3c329c51de60a183529d0c39ee3d86d9/programs/bpf_loader/src/serialization.rs